week3/homework/src/todo.js

```diff
@@ -0,0 +1,87 @@
+'use strict';
+
+const fs = require('fs');
+const uuid = require('uuid/v4');
+
+class Todo {
+  constructor(filename) {
+    this._filename = filename;
+  }
+
+  async create(description) {
+    const todos = await this.read();
+    const todo = {
+      id: uuid(),
+      done: false,
+      description,
+    };
+    todos.push(todo);
+    await this._save(todos);
+    return todo;
+  }
+
+  read() {
+    return new Promise(resolve => {
+      fs.readFile(this._filename, 'utf8', (error, data) => {
+        if (error) return resolve([]);
+        return resolve(JSON.parse(data));
+      });
+    });
+  }
+
+  async update(id, description) {
+    const todos = await this.read();
+
+    const todo = todos.find(t => t.id === id);
+    if (todo == null) {
+      const error = new Error(`To-do with ID ${id} does not exist`);
+      error.code = 'not-found';
+      throw error;
+    }
```

Comment:
the same code seems to be repeated in other functions. Any ideas on how to avoid repeating the code?